tornjak-frontend/src/components/spiffe-helper.js

```diff
@@ -114,64 +114,189 @@ class SpiffeHelper extends Component {
-      var lambdas = [];
-      var agentEntriesDict = {};
-
-      for (let i=0; i < agents.length; i++) {
-          let agent = agents[i];
-          let agentId = this.getAgentSpiffeid(agent);
-          agentEntriesDict[agentId] = [];
-
-          let agentSelectors = new Set(agent.selectors.map(formatSelectors));
-          let isAssocWithAgent = e => {
-              let entrySelectors = new Set(e.selectors.map(formatSelectors))
-              if (isSuperset(agentSelectors, entrySelectors)) {
-                  agentEntriesDict[agentId].push(e);
-              }
-          };
-          lambdas.push(isAssocWithAgent);
-      }
+  getAgentsEntries(agents, entries) {
+    let nodeEntries = entries.filter(e => e.parent_id.path === "/spire/server");
+    var lambdas = [];
+    var agentEntriesDict = {};
+
+    for (let i = 0; i < agents.length; i++) {
+      let agent = agents[i];
+      let agentId = this.getAgentSpiffeid(agent);
+      agentEntriesDict[agentId] = [];
 
-      for (let i=0; i < nodeEntries.length; i++) {
-          for (let j=0; j < lambdas.length; j++) {
-              lambdas[j](nodeEntries[i])
-          }
+      let agentSelectors = new Set(agent.selectors.map(formatSelectors));
+      let isAssocWithAgent = e => {
+        let entrySelectors = new Set(e.selectors.map(formatSelectors))
+        if (isSuperset(agentSelectors, entrySelectors)) {
+          agentEntriesDict[agentId].push(e);
+        }
+      };
+      lambdas.push(isAssocWithAgent);
+    }
+
+    for (let i = 0; i < nodeEntries.length; i++) {
+      for (let j = 0; j < lambdas.length; j++) {
+        lambdas[j](nodeEntries[i])
       }
+    }
 
-      return agentEntriesDict
+    return agentEntriesDict
   }
 
   // getCanonicalAgentSpiffeid takes in a agent entry, and returns the first 
   // agent ID that is found associated with it.
   getCanonicalAgentSpiffeid(entry, agents) {
-      let entrySelectors = new Set(entry.selectors.map(formatSelectors));
-      let isAssocWithEntry = a => {
-          let agentSelectors = new Set(a.selectors.map(formatSelectors));
-          return isSuperset(agentSelectors, entrySelectors);
+    let entrySelectors = new Set(entry.selectors.map(formatSelectors));
+    let isAssocWithEntry = a => {
+      let agentSelectors = new Set(a.selectors.map(formatSelectors));
+      return isSuperset(agentSelectors, entrySelectors);
+    }
+
+    let filteredAgents = agents.filter(isAssocWithEntry);
+    if (filteredAgents.length > 0) {
+      return this.getAgentSpiffeid(filteredAgents[0]);
+    }
+    return "";
+  }
+
+  // numberAgentEntries takes in spiffe id of an agent and entries list 
+  // returns number of entries for an agent
+  numberAgentEntries(spiffeid, globalEntries) {
+    if (typeof globalEntries !== 'undefined') {
+      var entriesList = globalEntries.filter(entry => {
+        return spiffeid === (this.getEntryParentid(entry))
+      })
+      return entriesList.length
+    } else {
+      return 0
```

Comment:
this is pretty much an error case, maybe we can return something like `NaN` here - i know it doesnt typecheck but at least it would show up eventually on the UI. May be helpful for us to catch things later.